http/logical.go

```diff
@@ -39,12 +39,53 @@ func (b *bufferedReader) Close() error {
 	return b.rOrig.Close()
 }
 
-func buildLogicalRequestNoAuth(perfStandby bool, w http.ResponseWriter, r *http.Request) (*logical.Request, io.ReadCloser, int, error) {
+func buildLogicalPathNoAuth(r *http.Request) (string, int, error) {
 	ns, err := namespace.FromContext(r.Context())
 	if err != nil {
-		return nil, nil, http.StatusBadRequest, nil
+		return "", http.StatusBadRequest, nil
+	}
+
+	path := r.URL.Path
+	if strings.HasPrefix(path, "/v1/") {
+		path = ns.TrimmedPath(path[len("/v1/"):])
```

Comment:
We want to always trim the namespace path regardless of the `v1/` presence so this should be outside of the prefix check.